releases, toc: add 3.1.0 GA release notes (#2296)
pingcap/docs: releases/3.1.0-ga.md

---
title: TiDB 3.1.0 GA Release Notes
category: Releases
---

# TiDB 3.1.0 GA Release Notes

Release date: April 16, 2020

TiDB version: 3.1.0 GA

TiDB Ansible version: 3.1.0 GA

## Compatibility Changes

+ TiDB

>     - Support directly stopping the start of TiDB if the HTTP listening port is unavailable, when the `report-status` configuration item is enabled [#16291](https://github.com/pingcap/tidb/pull/16291)

Reviewer: ```suggestion
    - Support directly stopping starting TiDB if the HTTP listening port is unavailable when the `report-status` configuration item is enabled [#16291](https://github.com/pingcap/tidb/pull/16291)
```

+ Tools

    - Backup & Restore (BR)

        * BR does not support restoring data from the TiKV cluster earlier than 3.1 GA [#233](https://github.com/pingcap/br/pull/233)

## New Features

+ TiDB

    - Support displaying the information of Coprocessor tasks in `explain format = "dot"` [#16125](https://github.com/pingcap/tidb/pull/16125)
    - Reduce the redundant stack information of log using the `disable-error-stack` configuration item [#16182](https://github.com/pingcap/tidb/pull/16182)

+ Placement Driver (PD)

    - Optimize the hot Region scheduling [#2342](https://github.com/pingcap/pd/pull/2342)

+ TiFlash

    - Add the metrics report related to the read and write workloads of DeltaTree engine
    - Support pushing down the `fromUnixTime` and `dateFormat` functions

>     - Disable the coarse-grained index filter by default

Author: ```suggestion
    - Disable the rough set filter by default
```

    - Fix the issue that warnings are mistaken as errors, and make the `UNION` statement have the same behavior as the `SELECT` statement [#16138](https://github.com/pingcap/tidb/pull/16138)
    - Fix the execution error when `TopN` is pushed down to mocktikv [#16200](https://github.com/pingcap/tidb/pull/16200)
    - Increase the initial length of `chunk.column.nullBitMap` to avoid unnecessary overhead of `runtime.growslice` [#16142](https://github.com/pingcap/tidb/pull/16142)

+ TiKV

    - Fix the panic issue caused by replica read [#7418](https://github.com/tikv/tikv/pull/7418) [#7369](https://github.com/tikv/tikv/pull/7369)
    - Fix the issue that the restoration process creates empty Regions [#7419](https://github.com/tikv/tikv/pull/7419)
    - Fix the issue that repeated resolve lock requests might harm the atomicity of pessimistic transactions [#7389](https://github.com/tikv/tikv/pull/7389)

+ TiFlash

    - Fix the potential issue of the `rename table` operation when replicating schema from TiDB

>     - Disable the coarse-grained index filter by default

Reviewer 2: ```suggestion
    - Disable the rough set filter by default
```

    - Fix the issue of data loss caused by the `rename table` operation under multiple data path configurations
    - Fix the issue that TiFlash reports incorrect storage space in some scenarios
    - Fix the potential issue caused by reading from TiFlash when Region Merge is enabled

+ Tools

    - TiDB Binlog

        * Fix the issue that TiFlash-related DDL jobs might interrupt the replication of Drainer [#948](https://github.com/pingcap/tidb-binlog/pull/948) [#942](https://github.com/pingcap/tidb-binlog/pull/942)

>     - BR

Reviewer 3: ```suggestion

    - BR

```
